crates/ntx-builder/src/actor/mod.rs

```diff
@@ -0,0 +1,339 @@
+///
+/// ## Lifecycle
+///
+/// 1. **Initialization**: Loads account state from the store or uses provided account data.
+/// 2. **Event Loop**: Continuously processes mempool events and executes transactions.
+/// 3. **Transaction Processing**: Selects, executes, and proves transactions, and submits them to
+///    block producer.
+/// 4. **State Updates**: Updates internal state based on mempool events and execution results.
+/// 5. **Shutdown**: Terminates gracefully when cancelled or encounters unrecoverable errors.
+///
+/// ## Concurrency
+///
+/// Each actor runs in its own async task and communicates with other system components through
+/// channels and shared state. The actor uses a cancellation token for graceful shutdown
+/// coordination.
+pub struct AccountActor {
+    origin: AccountOrigin,
+    store: StoreClient,
+    mode: ActorMode,
+    event_rx: mpsc::Receiver<MempoolEvent>,
+    cancel_token: CancellationToken,
+    block_producer: BlockProducerClient,
+    prover: Option<RemoteTransactionProver>,
+    chain_state: Arc<RwLock<ChainState>>,
+}
+
+impl AccountActor {
+    /// Constructs a new account actor and corresponding messaging channel with the given
+    /// configuration.
+    pub fn new(
+        origin: AccountOrigin,
+        config: &AccountActorConfig,
+        event_rx: mpsc::Receiver<MempoolEvent>,
+        cancel_token: CancellationToken,
+    ) -> Self {
+        let block_producer = BlockProducerClient::new(config.block_producer_url.clone());
+        let prover = config.tx_prover_url.clone().map(RemoteTransactionProver::new);
+        Self {
+            origin,
+            store: config.store.clone(),
+            mode: ActorMode::default(),
+            event_rx,
+            cancel_token,
+            block_producer,
+            prover,
+            chain_state: config.chain_state.clone(),
+        }
+    }
+
+    /// Runs the account actor, processing events and managing state until a reason to shutdown is
+    /// encountered.
+    pub async fn run(mut self, semaphore: Arc<Semaphore>) -> ActorShutdownReason {
+        // Load the account state from the store and set up the account actor state.
+        let account = {
+            match self.origin {
+                AccountOrigin::Store(account_prefix) => self
+                    .store
+                    .get_network_account(account_prefix)
+                    .await
+                    .expect("actor should be able to load account")
+                    .expect("actor account should exist"),
+                AccountOrigin::Transaction(ref account) => *(account.clone()),
+            }
+        };
+        let block_num = self.chain_state.read().await.chain_tip_header.block_num();
+        let mut state =
+            NetworkAccountState::load(account, self.origin.prefix(), &self.store, block_num)
+                .await
+                .expect("actor should be able to load account state");
```

Comment:
I would add a comment here about why we load the state even for accounts which have not been committed to the store yet because this is non-obvious.